code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -0,0 +1,199 @@
+			sleep(30)
+			src.death()
+	maxHealth = essence * 2
+	if(!revealed)
+		health = maxHealth //Heals to full when not revealed
+	if(essence_regen && !inhibited && essence < essence_regen_cap) //While inhibited, essence will not regenerate
+		essence++
+
+/mob/living/simple_animal/revenant/Process_Spacemove(var/check_drift = 0)
+	return 1 //Mainly to prevent the no-grav effect
+
+/mob/living/simple_animal/revenant/ex_act(severity)
+	return 1 //Immune to the effects of explosions.
+
+/mob/living/simple_animal/revenant/ClickOn(var/atom/A, var/params) //Copypaste from ghost code - revenants can't interact with the world directly.
+	if(client.buildmode)
+		build_click(src, client.buildmode, params, A)
+		return
+
+	var/list/modifiers = params2list(params)
+	if(modifiers["middle"])
+		MiddleClickOn(A)
+		return
+	if(modifiers["shift"])
+		ShiftClickOn(A)
+		return
+	if(modifiers["alt"])
+		AltClickOn(A)
+		return
+	if(modifiers["ctrl"])
+		CtrlClickOn(A)
+		return
+
+	if(world.time <= next_move)
+		return
+	A.attack_ghost(src)
+
+/mob/living/simple_animal/revenant/say(message)
+	return 0 //Revenants cannot speak out loud.
+
+/mob/living/simple_animal/revenant/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Current essence: [essence]E")
+
+/mob/living/simple_animal/revenant/New()
+	..()
+	spawn(5)
+		if(src.mind)
+			src << 'sound/effects/ghost.ogg'
+			src.store_memory("<span class='deadsay'>I am a revenant. My spectral form has been empowered. My only goal is to gather essence from the humans of [world.name].</span>")
+			src << "<br>"
+			src << "<span class='deadsay'><font size=3><b>You are a revenant!</b></font></span>"
+			src << "<b>Your formerly mundane spirit has been infused with alien energies and empowered into a revenant.</b>"
+			src << "<b>You are not dead, not alive, but somewhere in between. You are capable of very limited interaction with both worlds.</b>"
+			src << "<b>You are invincible and invisible to everyone but other ghosts. Some abilities may change this.</b>"
+			src << "<b>Your goal is to gather essence from humans. Your essence passively regenerates up to 25E over time. You can use the Harvest abilities to gather more from corpses.</b>"
+			src << "<b>Be sure to read the wiki page at https://tgstation13.org/wiki/Revenant !</b>"
+			src << "<br>"
+		if(!src.giveSpells())
+			message_admins("Revenant was created but has no mind. Trying again in five seconds.")
+			spawn(50)
+				if(!src.giveSpells())
+					message_admins("Revenant still has no mind. Deleting...")
+					qdel(src)
+
+/mob/living/simple_animal/revenant/proc/giveSpells()
+	if(src.mind)
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/targeted/revenant_harvest
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/targeted/revenant_transmit
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/aoe_turf/revenant_light
+		return 1
+	return 0
+
+/mob/living/simple_animal/revenant/death()
+	if(!strikes)
```

Comment:
Should this be `if(strikes)` instead? When strikes it zero it *should* be dying.